src/HealthChecks.AzureIoTHub/AzureIoTHubtHealthCheck.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices;
+
+namespace HealthChecks.AzureIoTHub
+{
+    public class AzureIoTHubHealthCheck : IHealthCheck
+    {
+        private readonly AzureIoTHubOptions _options;
+        public AzureIoTHubHealthCheck(AzureIoTHubOptions options)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                if (_options.RegistryWriteCheck)
+                {
+                    await ExecuteRegistryWriteCheckAsync(cancellationToken);
+                }
+                else if (_options.RegistryReadCheck)
+                {
+                    await ExecuteRegistryReadCheckAsync();
+                }
+                if (_options.ServiceConnectionCheck)
+                {
+                    await ExecuteServiceConnectionCheckAsync(cancellationToken);
+                }
+
+                return HealthCheckResult.Healthy();
+            }
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+        }
+
+        private async Task ExecuteServiceConnectionCheckAsync(CancellationToken cancellationToken)
+        {
+            var transportType = MapToTransportType(_options.ServiceConnectionTransport);
+            using (var client = ServiceClient.CreateFromConnectionString(_options.ConnectionString, transportType))
+            {
+                var result = await client.GetServiceStatisticsAsync(cancellationToken);
```

Comment:
rename result as statistics

Reply:
variable were removed as unused, enough to execute method (to do required) without any manipulations with `result/statistics`